Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/PropertyEditorToolsSystem.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Interface/Interface.h>
+#include <AzToolsFramework/UI/DocumentPropertyEditor/PropertyEditorToolsSystem.h>
+
+namespace AzToolsFramework
+{
+    PropertyEditorToolsSystem::PropertyEditorToolsSystem()
+    {
+        AZ::Interface<PropertyEditorToolsSystemInterface>::Register(this);
+    }
+
+    PropertyEditorToolsSystem::~PropertyEditorToolsSystem()
+    {
+        AZ::Interface<PropertyEditorToolsSystemInterface>::Unregister(this);
+    }
+
+    PropertyEditorToolsSystem::PropertyHandlerId PropertyEditorToolsSystem::GetPropertyHandlerForNode(const AZ::Dom::Value node)
+    {
+        using AZ::DocumentPropertyEditor::GetNodeName;
+        using AZ::DocumentPropertyEditor::Nodes::PropertyEditor;
+
+        if (!node.IsNode() || node.GetNodeName() != GetNodeName<PropertyEditor>())
+        {
+            AZ_Assert(false, "Attempted to look up a property handler for a value that is not a PropertyEditor node");
+            return InvalidHandlerId;
+        }
+
+        AZStd::string_view typeName = PropertyEditor::Type.ExtractFromDomNode(node).value_or("");
+        if (typeName.empty())
+        {
+            return InvalidHandlerId;
+        }
+
+        auto handlerBucketIt = m_registeredHandlers.find(AZ::Name(typeName));
+        if (handlerBucketIt == m_registeredHandlers.end())
+        {
+            return InvalidHandlerId;
+        }
+
+        for (auto handlerIt = handlerBucketIt->second.begin(); handlerIt != handlerBucketIt->second.end(); ++handlerIt)
+        {
+            if (handlerIt->m_shouldHandleNode(node))
+            {
+                return &(*handlerIt);
+            }
+        }
```

Comment:
Question: Are we still going to have the notion of default handlers vs. all registered handlers? In the current system we have a subset of handlers that will be automatically matched based on the value type, but others where the user has to explicitly request that handler by name. I will admit the previous system is a bit confusing, so I'd be on board with making it more straightforward, just wanted to see if this was an intentional design choice.

Reply:
Great question! Default handlers will be supported (in the form of a set of {type ID -> default handler name} mappings) but the plan is for those to be registered at the `AzFramework` level so that `DocumentAdapter` implementations (which have no concept of widgets) can query and use the default handler logic (e.g. the `ReflectionAdapter` will fill in default handlers). My tentative plan is to add a default handler registration option to `PropertyEditorSystemInterface` and automatically register the default RPE handlers to that.